assets/js/util/report-validation.test.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Tests for reporting API validation utilities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	isValidDateRange,
+	isValidDimensions,
+	isValidMetrics,
+	isValidOrders,
+} from './report-validation';
+
+describe( 'Reporting API validation', () => {
+	describe( 'isValidDateRange', () => {
+		it( 'should return TRUE if dateRange is valid only', () => {
+			expect( isValidDateRange( {
+				dateRange: 'last-14-days',
+				startDate: '',
+				endDate: null,
+			} ) ).toBeTruthy();
```

Comment:
```suggestion
			expect( isValidDateRange( {
				dateRange: 'last-14-days',
			} ) ).toBeTruthy();
```

Reading the test, those extra args make it _seem_ as if they're required but that `dateRange` being valid means they're allowed to be invalid… but the test here really is just that "passing a valid `dateRange` returns `true".

While the test is true, it ends up testing for not _exactly_ what it says, or at least combines a test of "`dateRange`-only works" with "valid `dateRange` + invalid `startDate`/`endDate` works".

Arguably we probably should _only_ accept one arg or the other, as combining them isn't clear as to which one takes precedence.

But at least here we should limit the arguments to clarify the test. Happy if you want to add extra tests like this to show that `dateRange` takes precedence over `startDate`+`endDate` if all are supplied, which seems to be the case, but at least I think we should limit the input for this test.

Reply:
> Happy if you want to add extra tests like this to show that dateRange takes precedence over startDate+endDate if all are supplied...

The `isValidDate Range` function has nothing to do with precedence, it just checks that either date range or start/end dates are provided.